### ISSUE TITLE
MyData first impressions

### ISSUE THREAD

**Issue by fititnt** [EticaAI/forum#62]
- https://mydata.org/
- http://sandbox.mydata.fi/

----

I recently found out about a project called MyData via twitter, and I started testing what I could find open source.

This was Logbook (step by step) https://github.com/fititnt/mydata-sdk--hiit/issues/1

**Comment by fititnt** [comment]
This was my first message at channel #general at 2018-04-18 10-18-00 BRT https://mydataglobal.slack.com/archives/C7G9FPL81/p1524964724000047.

I had some answers, but will not put here

----

I'm Emerson Rocha from Brazil. 

I'm a Full stack developer at my own company, and transdisciplinary researcher at Etica.AI.

Only in 2018 that I became part of international working groups possibly related to MyData. I am currently a member of _The IEEE Global Initiative on Ethics of Autonomous and Intelligent Systems_, _IEEE Society on Social Implications of Technology_ & _IEEE Special Interest Group on Humanitarian Technology (SIGHT)_.

I'm **not** an ethicist or data privacy expert, but I have great respect for those areas. I'm a **information technologist** expert. Here my GitHub https://github.com/fititnt.

I'm mostly concerned with strategies and open source software stacks near ready-to-use state that could allow ethical use of data and on future potentially allow design intelligent agents by non-experts. The focus is small to medium non governmental organizations who make social impact that normally are maintained by volunteers with low proficiency with technology.

I will suppose that most discussions here are more around big companies. I'm more concerned with _the other side of the spectrum_. As example, some of my network eventually are on regions with so poor internet access that the only way to go online is via WhatsApp (that yes, have unmetered data usage), so we look for _creative_ ways, like try to use the app as data proxy.

I'm new to the concept of the MyData, but I'm interested possible implementations of smaller groups (who **are self governed**) using both good theoretical approach already debated at international level and then implementation with open source stacks to allow direct access for other groups. For example: a regional association group for people with autism spectrum disorder decide track large amounts of sensitive personal information and, in addition of their on decisions based on data, have the option to share with other interests groups (like academic scientists or machine learning experts) and everyone have some set of rules and recommendations. A project more near this Citizen science (that do not had the sharing information part) is "Nightscout project" (the WeAreNotWaiting hashtag) & the Artificial pancreas.

Well, that was my introduction. My contact email is rocha@ieee.org.

**Comment by fititnt** [comment]
This was another message at 2018-04-30 18:46:00 BRT at channel #ourdata https://mydataglobal.slack.com/archives/C8SRFN6MC/p1525124773000219

----

Hi all. I presented myself on #general channel, and here will make some other comments.

I do not know who was behind this software decision, but *the container approach (docker) of mydata-sdk was a great move!* It not only make faster to test, but help in special who have lack of skilled devops personal on team.

If someone do not mentioned it yet, I would suggest go even further with the container approach. Like I said, I'm not ethicist/privacy expert, *but* I know about complex software stacks and strategies to abstract complexity.

> *First optimization (TD;DR: mydata-sdk already is very near to this)*

With some changes is possible to release using end user abstraction, for example is possible to make installable packages with Bitnami, kitematic, Portainer*, etc. The full mydata software stack could potentially run on a notebook computer to a non software developer person just clicking buttons (e.g. do not require the person do know command line commands), like other mainstream softwares that use these abstractions.

I will give a justification about why abstract more. Most people I have around who understand much more about privacy have no prior knowledge of devops, but they could be able to install on Wndows and Mac software like Wordpress/Joomla/Drupal/etc using these end user docker abstractions. So be able optionally to run the mydata SDK with some demos locally can make they more creative, and not only do force they to use some cloud central server (that, could not work if you have poor internet at time of presentation, or could be expensive) but also make they have lower fear of do something wrong and break for everyone.

> *Next optimization, use as demo software some most used CMS (TL;DR: needs extra code)*

The big players (Google, Amazon, Facebook and Microsoft, cited on mydata . org/what-we-want/) have good devops on their teams, but they still not represent most of the websites on internet. If we think about the rest of internet, they use open source. The top 3 are 30% Wordpress, 6% Joomla and 4% Drupal. These software and other ones already use these end user abstractions of containers and have documentation about how to create installable extensions (that could enable they to interact with mydata cores). If the mydata-sdk already easy for test, is possible to send some emails and some people of these communities can help to do the simple extensions.

Also, by have some basic plugins working on some already used CMSs, it's a step closer to be have a real usage with much fewer *perceived* extra moving parts. One example that does not require change laws is that some big governmental departments and some organizations with sensitive information (police, healthcare, etc) could **use a MyData core for audit own department/organization**. The user dashboard is not accessed by the user, but someone in his name: it could be a great audit tool with extra protection at infrastructure level when a specific software could have it's logs deleted or do not have a great log system but could push notices each time something important happens. But in this case, audit officers (internal or in special external ones) would love a extra feature on MyData core about alert in real time if very specific marked users have some specific events (_Quis custodiet ipsos custodes?_).

> I mean

*Is possible make a fully usable playground for end users with MyData able to run on a normal notebook, and (with some previous extra effort), be more near the end use by releasing as part of the demo package some of these more common CMSs.*


------
Some links:

- Bitnami: https://bitnami.com/stacks
- Kitematic: https://kitematic.com/
- Portainer: https://portainer.io/overview.html#demo (note: Portainer needs some extra configuration than Bitnami/Kitematic, but is possible to use it to manange with some much more friendly interface)

**Comment by fititnt** [comment]
Other recent post, since the Joomla project just made public this repository https://github.com/joomla-projects/privacy-framework


----

> Super recent news on a top open source CMS: The Joomla project made public just one hour ago the repository https://github.com/joomla-projects/privacy-framework/issues. One additional discussion is available here https://github.com / joomla / joomla-cms / issues / 20281. 

> Last Monday at https://mydataglobal.slack.com/archives/C8SRFN6MC/p1525124773000219 I commented about the idea of the potential of use go further on using docker containers and also release a version with some CMSs integrated with the MyData SDK, so non-developer could potentially run a semi-production version even on their notebooks.

> Also, the community that I grown up on open source was on the Joomla since ~2006 (software developer, forum moderator 2008-2014, community evangelist on Brazil, etc), so in this specific case I could help more than any other application.

> **So, did anyone here is interested in going deep on this and give attention**? mbabker is a hardcore developer, and he is leading the joomla team to rushing to release a complete API to allow all Joomla extensions comply with GDPR. As the Joomla 3.9 will be a LTS release, could make a lot of difference take action now to possibly make one of the most used CMSs in the world by default be pluggable with MyData SDK.

**Comment by fititnt** [comment]
At least on MyData group, I did not interacted much more, mostly because I was (and I am) busy with other things.

The repository I created with a fork to test their project (https://github.com/fititnt/mydata-sdk--hiit/issues) did not had other updates from me too. The main repositpory https://github.com/HIIT/mydata-sdk was archived with no new updates. (just to remember: My PR at https://github.com/HIIT/mydata-sdk/pull/9/files asking to warn the users to point to https://github.com/mydata-sdk/mydata-sdk took some time but was done in repository description at the top. Good thing.)

I will close this issue. Even if have something new, would be not more "first impressions". But my impression with they and all their work was nice, in special considering that is a big project.
